test/directed-requester.js

```diff
@@ -0,0 +1,91 @@
+import test from 'ava';
+import r from 'randomstring';
+import LogSuppress from '../lib/log-suppress';
+
+LogSuppress.init(console);
+
+const environment = r.generate();
+const { DirectedRequester, Responder } = require('../')({ environment });
+
+test.cb('Supports directed requests using callbacks', (t) => {
+    const key = r.generate();
+    const namespace = r.generate();
+
+    const requester = new DirectedRequester(
+        { name: `PBR ${t.title}: directed requester`, key, namespace });
+    const responders = [];
+
+    Array.from({ length: 10 }).forEach(() => {
+        responders.push(new Responder({ name: `PBR ${t.title}: directed responder`, key, namespace }));
+    });
+
+    let randomResponder = responders[Math.floor(Math.random() * responders.length)];
+
+    randomResponder.sock.on('bind', () => {
+        const args = [1, 2, 6];
+        requester.send({ responderId: randomResponder.discovery.me.id, type: 'test', args }, (res) => {
+            t.deepEqual(res, args);
+            t.end();
+        });
+
+        randomResponder.on('test', (req, cb) => {
+            t.deepEqual(req.args, args);
+            cb(req.args);
+        });
+    });
+});
+
+test.cb('Supports directed requests using promises', (t) => {
```

Comment:
Please also update this test with the deterministic approach laid out in my first comment.